src/ai/backend/manager/models/base.py

```diff
@@ -660,6 +661,37 @@ def from_row(
     ) -> _GenericSQLBasedGQLObject: ...
 
 
+class SlugType(TypeDecorator):
+    """
+    A type wrapper for allowing unicode
+    """
+
+    impl = sa.types.Unicode
+    cache_ok = True
+
+    def __init__(self, *, length: int = 64, allow_unicode: bool = False) -> None:
+        self._allow_unicode = allow_unicode
+        self._length = length
+        self._rx_slug = re.compile(r"^[a-zA-Z0-9]([a-zA-Z0-9._-]*[a-zA-Z0-9])?$")
```

Comment:
How about using the `\w` pattern in place of `[a-zA-Z0-9]` and setting `re.ASCII` flag as the inverse of `allow_unicode`?